### PR TITLE
For #28273 - Configure search with the selected search engine before user searching

### LINKED ISSUE #28273: [Bug] Unified search: First homescreen search does not properly differentiate between search engines
Steps to reproduce:
- have unified search enabled
- have the homescreen showing
- tap on the unified search selector to choose a different search engine
- choose any other than the default (easier to see the issue with application engines like tabs/bookmarks/history)

Expected result:
- the next search show results particular to that search engine

Actual result:
- the next search shows are not correctly filtered

Issues affects all app versions.

Video showing the initially incorrect search results and then the expected search results after circling through search engines:

https://user-images.githubusercontent.com/11428869/208849770-10966532-5207-440b-8e10-25aba61c9edd.mp4

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-22178)

### PR DESCRIPTION
**Pull request by Mugurell**
https://user-images.githubusercontent.com/11428869/209098478-c59de00b-4c1f-4bb1-8d28-49f94bad6c1c.mp4

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #28273

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt [hunk 1]
```diff
@@ -266,6 +266,15 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
 
         requireComponents.core.engine.speculativeCreateSession(isPrivate)
 
+        // Handle the scenario in which the user selects another search engine before starting a search.
```
Comment (t-p-white) [inline app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt:269]: nit: this logic could be moved to a func which would remove the need for the suppression and potentially self document

**Review by t-p-white — COMMENTED** [review]
Apart from the nit, lgtm 👍

**Review by t-p-white — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt (hunk 1 above)
Comment (Mugurell) [reply app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt:269]: Thank you!
Extracted the code to a separate method which also ensures easier testing. Would prefer to keep the current comment as context for why we call that method.

Comment (t-p-white) [reply app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt:269]: Understood 👍 Appreciate the added tests

**Review by t-p-white — APPROVED** [review]

**Pull request merged**